libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -224,9 +224,6 @@ public SyncState reSync(long syncId, SyncUpdateCallback callback) throws JSONExc
         if (sync.getType() != SyncState.Type.syncDown) {
             throw new SmartSyncException("Cannot run reSync:" + syncId + ": wrong type:" + sync.getType());
         }
-        if (sync.getTarget().getQueryType() != SyncTarget.QueryType.soql) {
```

Comment:
Re-sync should not be limited to one type of sync down (since custom sync target can support resync as well).